### PR TITLE
fix: fix create table error but table id uncommttied

### LINKED ISSUE #15664: bug: create table error make alter table fail
### Search before asking

- [X] I had searched in the [issues](https://github.com/datafuselabs/databend/issues) and found no similar issues.

### Version

v1.2.497-nightly

### What's Wrong?

```sql
CREATE OR REPLACE TABLE test_table as select 1;
CREATE OR REPLACE TABLE test_table as select 1/0;
alter table test_table rename to test_table_1;   // will fail

select * from test_table;  // but select works!
```

### How to Reproduce?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by lichuang**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

fix: fix create table error but table id uncommttied

when create table use `as select`, it will save table id in a temporary table id list, if create table success, table id will be commit, otherwise table id\meta\s3 data will be vacuumed when `vacuum drop table`.

- Fixes #15664

## Tests

- [ ] Unit Test
- [x] Logic Test
- [ ] Benchmark Test
- [ ] No Test  - _Explain why_

## Type of change

- [x] Bug Fix (non-breaking change which fixes an issue)
- [ ] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/15665)

### CODE REVIEW

File: src/query/service/src/interpreters/interpreter_table_create.rs [hunk 1]
```diff
@@ -273,9 +274,28 @@ impl CreateTableInterpreter {
...
+                            table_id,
+                            table_name,
+                        };
+                        catalog.rollback_uncommitted_table_meta(drop_by_id).await
+                    };
+
+                    GlobalIORuntime::instance().block_on(fut).map_err(|e| {
+                        info!("create {} as select failed. {:?}", qualified_table_name, e);
+                        e
+                    })?;
```
Comment (dantengsky) [inline src/query/service/src/interpreters/interpreter_table_create.rs:295]: if process crashed before `rollback_uncommitted_table_meta` successfully executed, will things still work as expected?

**Comment by dantengsky** [comment]
@lichuang 

maybe, we could also consider tweaking the following snippet of ` SchemaApi for KV::rename_table`?

instead of assuming that `last_table_id == table_id` and pop it out,  find the table_id and remove it from `tb_id_list`?

https://github.com/datafuselabs/databend/blob/3109054a4dfea22009026479bd9a21d87eff13e3/src/meta/api/src/schema_api_impl.rs#L1876-L1892

**Comment by drmingdrmer** [comment]
> @lichuang
> 
> maybe, we could also consider tweaking the following snippet of ` SchemaApi for KV::rename_table`?
> 
> instead of assuming that `last_table_id == table_id` and pop it out, find the table_id and remove it from `tb_id_list`?
> 
> https://github.com/datafuselabs/databend/blob/3109054a4dfea22009026479bd9a21d87eff13e3/src/meta/api/src/schema_api_impl.rs#L1876-L1892

Table list is a stack which is responsible to pop up the last removed table.

**Comment by dantengsky** [comment]
For your reference: 

another way to resolve issue #15664

https://github.com/datafuselabs/databend/pull/15676

**Pull request merged**